core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -112,7 +115,20 @@ class SocketServer(val config: KafkaConfig, val metrics: Metrics, val time: Time
         }.sum / totalProcessorThreads
       }
     )
-
+    val memoryPoolDepletedPercentMetricName = metrics.metricName("MemoryPoolAvgDepletedPercent-Avg", "socket-server-metrics")
+    val memoryPoolSensor = metrics.sensor("MemoryPoolAvgDepletedPercent")
+    memoryPoolSensor.add(memoryPoolDepletedPercentMetricName, new Rate(TimeUnit.MILLISECONDS))
```

Comment:
the MemoryPool interface and implementations are in clients, which has no dep. on yammer.
i could either add a dep on yammer (probably bad idea) or introduce an intermediary interface ?